### DISCUSSION TITLE
CommonRoad version of XML-file is not supported

### DISCUSSION THREAD

**Discussion by lizhenghao2** [CommonRoad/commonroad-io#1] [Q&A] [answered]
I tried to get motion planning solutions from a docker submission provided by former competitioners, I got the solutions successfully but was not able to read them. The version of commonroad-io inside the container created fromo the docker submission is 2022.1, I built a conda envirionment(cr-io version: 2023.4) outside the container to read the results, I used the code example provided in commonroad-io gir repository to read them, I had no problem reading and visualizing the scenarios, but when reading the solutioins, the following error pops out:

Traceback (most recent call last):
File “code_example_2022.py”, line 10, in
scenario, planning_problem_set = CommonRoadFileReader(file_path).open()
File “/home/uwu/anaconda3/envs/io-202201/lib/python3.7/site-packages/commonroad/common/file_reader.py”, line 81, in open
self._read_header()
File “/home/uwu/anaconda3/envs/io-202201/lib/python3.7/site-packages/commonroad/common/file_reader.py”, line 127, in _read_header
commonroad_version)
AssertionError: <CommonRoadFileReader/_read_header>: CommonRoad version of XML-file solutions_example/solution_KS1:TR1:DEU_Aachen-3_1_I-1:2020a.xml is not supported. Supported versions: {‘2018b’, ‘2020a’}. Got version: None.

Also I'm not sure how to read and visualize the the solutions for interactive scenarios, is there a code example for it?

**Comment by smaierhofer** [comment, marked as answer]
It looks like you are trying to read a solution file with the scenario file reader.
You have to use the [CommonRoadSolutionReader](https://github.com/CommonRoad/commonroad-io/blob/1d45b9bb6a6fbb90ae65725bf006f84d2f1eddca/commonroad/common/solution.py#L638) instead.
